lib/Sema/TypeCheckStorage.cpp

```diff
@@ -1968,6 +1968,17 @@ static AccessorDecl *createSetterPrototype(AbstractStorageDecl *storage,
   
   // Copy availability from the accessor we'll synthesize the setter from.
   SmallVector<Decl *, 2> asAvailableAs;
+
+  auto addEnclosingScopeAvailability = [&]() {
```

Comment:
This looks similar to the code in configureInheritedDesignatedInitAttributes(). Maybe applyInferredAvailableAttrs() should take an optional Decl in addition to the array of AvailableAttrs, and walk over the Decls parents if its not null?

Reply:
Actually it might be better to extract a getInnermostAvailabilityOfDecl() or something and use it in both places, or perhaps its not worth it at all...

Reply:
I think it'd be useful to extract this code somewhere, especially if the availability inference doesn't work for `_modify` (which I think it doesn't), so it'll need to be used there as well